### PR TITLE
Enable TLS 1.3 post-handshake authentication

### PR DESCRIPTION
**Pull request by tiran**
Fixes: https://github.com/urllib3/urllib3/issues/1634
Signed-off-by: Christian Heimes <christian@python.org>

### CODE REVIEW

**Comment by sethmlarson** [comment]
Thanks for this! Could you add a changelog entry for this change and a simple test that shows that this value is either being set to `True` or is not an attribute for the `SSLContext`? I'm not sure if we can easily test the functionality within our current test suite.

**Comment by tiran** [comment]
There is currently a bug in either OpenSSL or Python that causes a side effect when PHA is enabled: https://bugs.python.org/issue37428

Let me first sort out the bug, then we can land this PR.

**Comment by stratakis** [comment]
https://bugs.python.org/issue37428 has been resolved on the python side.

**Comment by tiran** [comment]
@sethmlarson I have rebased the PR and added an additional test case for PHA. Let's see of AppVeyor passes now.

**Comment by pquentin** [comment]
AppVeyor fails because of an unrelated issue (https://github.com/urllib3/urllib3/issues/1664), and Travis fails because you now need to run `nox -s blacken`. So this change looks good!

**Comment by sethmlarson** [comment]
Thanks for this @tiran! I'm going to merge it since the falling AppVeyor isn't related.

File: test/test_ssl.py [hunk 1]
```diff
@@ -149,3 +149,18 @@ def test_wrap_socket_default_loads_default_certs(monkeypatch):
     ssl_.ssl_wrap_socket(sock)
 
     context.load_default_certs.assert_called_with()
+
+
+@pytest.mark.parametrize(
+    ["pha", "expected_pha"], [(None, None), (False, True), (True, True)]
```
Comment (pquentin) [inline test/test_ssl.py]: @tiran We were wondering with @jab in https://github.com/python-trio/urllib3/pull/127: any reason why setting PHA to False sets it back to True? In other words, why `(False, True)` and not `(False, False)`?

**Pull request merged**